CONTRIBUTING.md

```diff
@@ -1,60 +1,145 @@
-# Typical development workflow
+# Contributing
 
-Fork the repo, work on an issue
+This guide explains how to set up your environment, make changes, and submit a PR.
 
-## Updating the generated Kibana client.
+## Development Setup
 
-If your work involves the Kibana API, the endpoints may or may not be included in the generated client.
-Check [generated/kbapi](./generated/kbapi/) for more details.
+* Fork and clone the repo.
+* Setup your preferred IDE (Intelliji, VSCode, etc.)
 
-## Acceptance tests
+Requirements:
+* [Terraform](https://www.terraform.io/downloads.html) >= 1.0.0
+* [Go](https://golang.org/doc/install) >= 1.25
+* Docker (for acceptance tests)
 
-```bash
-make docker-testacc
-```
+## Development Workflow
 
-Run a single test with terraform debug enabled:
-```bash
-env TF_LOG=DEBUG make docker-testacc TESTARGS='-run ^TestAccResourceDataStreamLifecycle$$'
-```
+* Create a new branch for your changes.
+* Make your changes. See [Useful Commands](#useful-commands) and [Debugging](#debugging).
+  * GitHub Copilot can be used to help with these changes (See [Using Copilot](#using-copilot))
+* Validate your changes
+  * Run unit and acceptance tests (See [Running Acceptance Tests](#running-acceptance-tests)).
+  * Run `make fmt`, `make lint`.
+  * All checks also run automatically on every PR.
+* Add a changelog entry in `CHANGELOG.md` under the `Unreleased` section. This will be included in the release notes of the next release.
+* Submit your PR for review.
 
-A way to forward debug logs to a file:
-```bash
-env TF_ACC_LOG_PATH=/tmp/tf.log TF_ACC_LOG=DEBUG TF_LOG=DEBUG make docker-testacc
-```
+When creating new resources:
+* Use the [Plugin Framework](https://developer.hashicorp.com/terraform/plugin/framework/getting-started/code-walkthrough) for new resources.
+  * Use an existing resource (e.g. `internal/elasticsearch/security/system_user`) as a template.
+  * Some resources use the deprecated Terraform SDK, so only resources using the new Terraform Framework should be used as reference.
+* Use the generated API clients to interact with the Kibana APIs. (See [Working with Generated API Clients](#working-with-generated-api-clients)
+* Add documentation and examples for the resource. Update the generated docs with `make docs-generate`.
```

Comment:
We also need to add the doc template ([example](https://github.com/elastic/terraform-provider-elasticstack/blob/main/templates/resources/elasticsearch_security_system_user.md.tmpl)) as part of this work. 

The docs generator doesn't (or didn't, I haven't checked recently) populate the `subcategory` field in the frontmatter which we're using to improve the docs organisation.